OpenEdXMobile/src/main/java/org/edx/mobile/base/MainApplication.java

```diff
@@ -133,24 +133,29 @@ private void init() {
     private void checkIfAppVersionUpgraded(Context context) {
         PrefManager.AppInfoPrefManager prefManager = new PrefManager.AppInfoPrefManager(context);
         long previousVersionCode = prefManager.getAppVersionCode();
-        final long currVersionCode = (long) BuildConfig.VERSION_CODE;
-        if (previousVersionCode < currVersionCode) {
+        final long curVersionCode = (long) BuildConfig.VERSION_CODE;
```

Comment:
You can keep the type of the variable to `long` for similarity, but still the cast is unnecessary cuz upward casts are always implicit.